crypto/ml_kem/ml_kem_ipd_ref_common/cbd.c

```diff
@@ -105,17 +101,15 @@ static void cbd3(poly *r, const uint8_t buf[3*KYBER_N/4])
     }
   }
 }
-#endif
 
-void poly_cbd_eta1(poly *r, const uint8_t buf[KYBER_ETA1*KYBER_N/4])
+void poly_cbd_eta1(ml_kem_params *params, poly *r, const uint8_t *buf)
 {
-#if KYBER_ETA1 == 2
-  cbd2(r, buf);
-#elif KYBER_ETA1 == 3
-  cbd3(r, buf);
-#else
-#error "This implementation requires eta1 in {2,3}"
-#endif
+  assert((params->eta1 == 2) || (params->eta1 == 3));
+  if (params->eta1 == 2) {
+    cbd2(r, buf);
+  } else {
+    cbd3(r, buf);
+  }
 }
 
 void poly_cbd_eta2(poly *r, const uint8_t buf[KYBER_ETA2*KYBER_N/4])
```

Comment:
Don't need pre-processing in this function

Reply:
I wanted to minimize changes in this PR and change only what is necessary so I skipped things like this. I'd really rather leave this as is and refactor/reimplement once we get more time allocated for it.